Kyrychenko/quizEntity/src/main/java/ui/AuthenticationServlet.java

```diff
@@ -0,0 +1,100 @@
+package ui;
+
+import model.User;
+import service.UserService;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.util.Arrays;
+
+@WebServlet("/authentication")
+public class AuthenticationServlet extends HttpServlet {
+    private static final String PARAM_NAME_USERNAME = "name";
+    private static final String PARAM_NAME_LOGIN = "login";
+    private static final String PARAM_NAME_PASSWORD = "password";
+
+    private static final String LOGIN_PAGE = "/login.jsp";
+    private static final String REGISTRATION_PAGE = "/registration.jsp";
+
+    private static final String ERROR_EMPTY_LOGIN_PASS_MESSAGE = "One of the entered values are empty";
+    private static final String ERROR_LOGIN_PASS_MESSAGE = "Incorrect login or password";
+    private static final String ERROR_LOGIN_EXIST_MESSAGE = "User with such login already exist";
+
+    private static UserService userService = UserService.getInstance();
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        HttpSession session = req.getSession();
+
+        String login = req.getParameter(PARAM_NAME_LOGIN);
+        String password = req.getParameter(PARAM_NAME_PASSWORD);
+
+        if (isEmptyParameters(login, password)) {
+            forwardWithErrorMessage(req, resp, LOGIN_PAGE,  ERROR_EMPTY_LOGIN_PASS_MESSAGE);
+        } else if (!userService.isUserAccountFound(login, password)) {
+            forwardWithErrorMessage(req, resp, LOGIN_PAGE, ERROR_LOGIN_PASS_MESSAGE);
+        } else {
+            session.setAttribute("userID", session.getId());
+            session.setAttribute("login", login);
+            resp.sendRedirect("/quiz");
+        }
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        HttpSession session = req.getSession();
```

Comment:
I'd better separate registration, login and logout into three servlets and three different endpoints. Why do we need this if/else/action game, if they are different actions at all?